File: Unit2-04/Typescript/MrCoxallStack.ts
/**
* This is the MrCoxallStack class
*
* By:      Evan Beaudoin
* Version: 1.0
* Since:   2024-04-01
*/ 

export class MrCoxallStack {
  // Stack that contains all of the string
  private _stackAsList: string[]

  // Creates the stack
  public constructor() {
    this._stackAsList = []
  }

  // Getter for stackAsList
  public get showStack() {
      let stackItems = ""

      for (let counter = 0; counter < this._stackAsList.length; counter++) {
        // If statments ensures that comma is not printed at the start
        if (counter === 0) {
          stackItems += this._stackAsList[counter]
        } else {
          stackItems += ", " + this._stackAsList[counter]
        }
      }

      return stackItems
    }

  // Checks if the stack is empty
  public get isEmpty() {
  let empty = false
    if (this._stackAsList.length == 0) {
      empty = true
    }
    return empty
  }
  // returns number of elemnts in stack
  public get size() {
      return this._stackAsList.length
  }

  // Pushes the values to the stack
  public push(stringPushed) {
    this._stackAsList.push(stringPushed)
  }

  public popItem() {
    let returnValue = ""

    if (this._stackAsList.length > 0) {
      returnValue = this._stackAsList.pop()
    }
    else {
      console.log("Array is empty!")
    }
    return returnValue
  } 

  public peekItem() {
    let returnValue = ""

    if (this._stackAsList.length > 0) {
      returnValue = this._stackAsList[this._stackAsList.length - 1]
    }
    else {
      console.log("Array is empty")
    }
    return returnValue
  }
  public emptyStack() {
    this._stackAsList = ""
    return this._stackAsList
  }
}

File: Unit2-04/Java/MrCoxallStack.java
/*
* This class defines a string stack
*
* @author  Evan Beaudoin
* @version 1.0
* @since   2024-04-01
*/

import java.util.ArrayList;
import java.util.List;

/**
 * This is the class.
*/
public class MrCoxallStack {
    /**
     * Adds strings to an array.
     *
     * @param array contains the array of strings
    */
    private List<String> stackAsList;

    /**
     * Constructs a new instance of MrCoxallStack.
     * Initializes the stack with an empty ArrayList.
     *
    */
    public MrCoxallStack() {
        this.stackAsList = new ArrayList<>();
    }

    /**
    * Returns the stack value.
    *
    * @return the values of the list formatted
    */
    public String showStack() {
        final StringBuilder stackValues = new StringBuilder();
        // https://www.javatpoint.com/StringBuilder-class

        for (String value : this.stackAsList) {
            stackValues.append(value).append(", ");
        }

        stackValues.deleteCharAt(stackValues.length() - 2);

        return stackValues.toString();
    }

    /**
     * Checks if the array is empty.
     *
     * @return the bool that says whether the array is empty or not
    */
    public boolean isEmpty() {
        boolean empty = false;

        if (this.stackAsList.size() == 0) {
            empty = true;
        }

        return empty;
    }

    /**
     * Checks the size of the array.
     *
     * @return the array's length
    */
    public int size() {
        return this.stackAsList.size();
    }

    /**
     * Pushes the string to a list.
     *
     * @param string the string that is going to be pushed
    */
    public void push(String string) {
        this.stackAsList.add(string);
    }

    /**
     * Returns the stack value.
     *
     * @return the values of the list formatted
     *
     * @throws IllegalStateException if this is ever called
    */
    public String pop() {
        if (stackAsList.isEmpty()) {
            throw new IllegalStateException("stack is empty.");
        }

        return stackAsList.remove(stackAsList.size() - 1);
    }

    /**
     * Returns the last stack value.
     *
     * @return the last value.
     *
     * @throws IllegalStateException if this is ever called
    */
    public String peekItem() {
        if (stackAsList.isEmpty()) {
            throw new IllegalStateException("Stack is empty.");
        }

        return stackAsList.get(stackAsList.size() - 1);
    }

    /**
     * Clears the stack.
    */
    public void emptyStack() {
        this.stackAsList.clear();
    }
}
